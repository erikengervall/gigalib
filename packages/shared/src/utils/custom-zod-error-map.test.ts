import { z } from 'zod';

import {
  customZodErrorMap,
  resetGlobalCustomZodErrorMap,
  setGlobalCustomZodErrorMap,
} from './custom-zod-error-map';

const Person = z.object({
  name: z.string(),
});

describe('customZodErrorMap', () => {
  it('should set the custom zod error map', () => {
    // with default error map
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();

    // with custom error map
    setGlobalCustomZodErrorMap();
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();

    // reset -> back to default error map
    resetGlobalCustomZodErrorMap();
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();

    // with custom error map (without data)
    setGlobalCustomZodErrorMap({ appendInputData: false });
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();

    // reset -> back to default error map
    resetGlobalCustomZodErrorMap();
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();

    // with custom error map (with data)
    expect(() =>
      Person.parse(
        { name: 123 },
        { errorMap: customZodErrorMap({ appendInputData: true }) },
      ),
    ).toThrowErrorMatchingSnapshot();

    // the custom error map should not be set globally
    expect(() => Person.parse({ name: 123 })).toThrowErrorMatchingSnapshot();
  });
});
