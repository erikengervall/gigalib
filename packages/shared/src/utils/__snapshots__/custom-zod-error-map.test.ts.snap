// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customZodErrorMap should set the custom zod error map 1`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 2`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number [DATA]<123>"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 3`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 4`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 5`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 6`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number [DATA]<123>"
  }
]"
`;

exports[`customZodErrorMap should set the custom zod error map 7`] = `
"[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      "name"
    ],
    "message": "Expected string, received number"
  }
]"
`;
